### ISSUE TITLE
aws_ecs_patterns: ScheduledFargateTask (and others)

### ISSUE THREAD

**Issue by mrtngrbmn** [aws/aws-cdk#27517]
### Describe the issue

In aws_ecs_pattern.ScheduledFargateTask there are
- scheduled_fargate_task_definition_options 
- scheduled_fargate_task_image_options 
- task_definition 

Difference between the first is clear, but how is the task_definition parameter is related to them?
What is the meaning of "[disable-awslint:ref-via-interface] Default: - none"?

In aws_ecs_patterns.ScheduledFargateTaskDefinitionOptions:
"Image or taskDefinition must be specified, but not both. [disable-awslint:ref-via-interface]" is stated, what makes no sense here.

In aws_ecs_patterns.ScheduledFargateTaskImageOptions:
The parameter 
"task_definition (Optional[[FargateTaskDefinition](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ecs/FargateTaskDefinition.html#aws_cdk.aws_ecs.FargateTaskDefinition)]) – The task definition to use for tasks in the service. TaskDefinition or TaskImageOptions must be specified, but not both. [disable-awslint:ref-via-interface] Default: - none"
is not clear to me and how it make sense here.

### Links

https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ecs_patterns/ScheduledFargateTask.html
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ecs_patterns/ScheduledFargateTaskImageOptions.html
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ecs_patterns/ScheduledFargateTaskDefinitionOptions.html

**Comment by peterwoodworth** [comment]
There were warnings added to call this out - are you seeing these warnings? https://github.com/aws/aws-cdk/pull/26737

**Comment by mrtngrbmn** [comment]
No. I'm using ScheduledFargateTaskImageOptions with some parameters and the task_definition parameter. There is no warning, neither for  cdk synth nor for cdk deploy.

**Comment by peterwoodworth** [comment]
Well, the warning will pop up if you use the high level `taskDefinition` parameter. This was done because the `taskDefinition` prop originates from extending another construct, but on this higher level construct it's redundant. However the docs generation won't know that and know how to give it a different docstring, which is why the warning was added

If the difference between the first two is clear, then that's as good as we can do here in terms of documentation

**Comment by mrtngrbmn** [comment]
Difference between the first two is clear. Just a request to update the documentation.
